.github/actions/winget/index.js

```diff
@@ -0,0 +1,112 @@
+const yaml = require('js-yaml');
+const exec = require('@actions/exec');
+const core = require("@actions/core");
+const glob = require('@actions/glob');
+const path = require('path');
+const compareVersions = require('compare-versions');
+const fs = require('fs/promises');
+const fsSync = require('fs');
+const io = require('@actions/io');
+const crypto = require('crypto');
+const { https } = require('follow-redirects');
+
+async function run() {
+    const token = core.getInput('token', {
+        required: true,
+        trimWhitespace: true
+    });
+    let newVersion = core.getInput('version', {
+        required: true,
+        trimWhitespace: true
+    });
+    const idx = newVersion.lastIndexOf('-');
+    if (idx !== -1)
+        newVersion = newVersion.substr(0, idx);
+    const assets = JSON.parse(core.getInput('assets', {
+        required: true,
+    }));
+    let x86Url, x64Url;
+    for (const data of assets) {
+        if (data.browser_download_url.endsWith('.exe')) {
+            if (data.browser_download_url.includes('x86'))
+                x86Url = data.browser_download_url;
+            else
+                x64Url = data.browser_download_url;
+        }
+    }
+
+    await exec.exec('git', ['clone', `https://github.com/microsoft/winget-pkgs.git`]);
+    const globber = await glob.create('.\\winget-pkgs\\manifests\\e\\eloston\\ungoogled-chromium\\*', {matchDirectories: true, implicitDescendants: false});
+    const pathList = await globber.glob();
+    const ucPath = path.dirname(pathList[0]);
+    const versionList = pathList.map(x => path.basename(x));
+    versionList.sort(compareVersions);
+    const latestVersion = versionList[versionList.length - 1];
+    const latestVersionPath = path.join(ucPath, latestVersion);
+    const newVersionPath = path.join(ucPath, newVersion);
+    try {
+        await io.mkdirP(newVersionPath);
+    } catch (e) {
+    }
+
+    await updateInstaller(latestVersionPath, newVersionPath, latestVersion, newVersion, x86Url, x64Url);
+    await replaceContent(latestVersionPath, newVersionPath, latestVersion, newVersion, 'eloston.ungoogled-chromium.locale.en-US.yaml');
+    await replaceContent(latestVersionPath, newVersionPath, latestVersion, newVersion, 'eloston.ungoogled-chromium.yaml');
+
+    await new Promise((resolve, reject) => {
+        const file = fsSync.createWriteStream('wingetcreate.exe');
+        https.get('https://aka.ms/wingetcreate/latest', resp => {
```

Comment:
Not a deal breaker but in the future this can probably be replaced with `fetch` (once it gets default included in node)